monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,324 @@
+
+        ''' Check if atleast one track is provided, if no tracks are provided, then RIDAircraftState and Test JSON cannot be generated.''' 
+
+        if (self.flight_points): # Empty flight points cannot be converted to a Aircraft State, check if the list has atleast one value
+            return True
+        else: 
+            raise ValueError("At least one flight track is necessary to create a AircraftState and a test JSON, please generate the tracks first using AdjacentCircularFlightPathsGenerator class")
+
+    def write_rid_state(self, duration = 180):
+
+        ''' This method iterates over flight tracks and geneates AircraftState JSON objects and writes to disk in the test_definitions folder, these files can be used to submit the data in the test harness '''
+
+        flight_lenghts = {} # Develop a index of flight length and their index
+        flight_current_index = {} # Store where on the track the current index is, since the tracks are circular, once the end of the track is reached, the index is reset to 0 to indicate beginning of the track again. 
+        num_flights = len(self.flight_points) # Get the number of flights 
+        now = arrow.now() 
+        flight_telemetry = {}
+        for i in range(num_flights):
+            flight_lenghts[i]= len(self.flight_points[i])
+            flight_current_index[i] = 0
+            flight_telemetry[i] = []
+        
+        for j in range(duration):
+            if j == 0:
+                timestamp = now.shift(seconds = 2)
+            else:
+                timestamp = timestamp.shift(seconds = 2)
+            seconds_diff = (now - timestamp).total_seconds()
+            
+            for k in range(num_flights):
+                list_end = flight_lenghts[k] - flight_current_index[k]            
+                if list_end != 1:             
+                    flight_point = self.flight_points[k][flight_current_index[k]]
+                    aircraft_position = AircraftPosition(lat = flight_point.lat, lng = flight_point.lng, alt = flight_point.alt, accuracy_h= "HAUnkown", accuracy_v = "VAUnknown", extrapolated = 1, pressure_altitude = 0)
+
+                    rid_aircraft_state = {'id':k, "aircraft_type":"NotDeclared", "current_state":{"timestamp": seconds_diff,"operational_status":"Undeclared", "position":{"lat":aircraft_position.lat, "lng":aircraft_position.lng, "alt":aircraft_position.alt, "accuracy_h": aircraft_position.accuracy_h, "accuracy_v":aircraft_position.accuracy_v, "extrapolated":aircraft_position.extrapolated, "pressure_altitude": aircraft_position.pressure_altitude}, "height":{"distance":0,"reference": "TakeoffLocation" },"track":0,"speed":1.9, "speed_accuracy":"SAUnknown", "vertical_speed":0.2,"group_radius":0, "group_ceiling": 0, "group_floor": 0, "group_count": 1, "group_time_start": seconds_diff, "group_time_end": seconds_diff}} # In the RID Aircraft state object the only object to update is the "seconds_diff" in the rid_aircraft_state
```

Comment:
I have added a new `reference_time` which has the `now` parameter while I like the timedelta field, adding it will mean that the `rid_aircraft_state` will have two fields that are not in the standard. I have kept the timestamp but it might be useful to have a the timedelta anyway?